op-batcher/batcher/channel_manager.go

```diff
@@ -426,82 +416,93 @@ func l2BlockRefFromBlockAndL1Info(block *types.Block, l1info *derive.L1BlockInfo
 
 var ErrPendingAfterClose = errors.New("pending channels remain after closing channel-manager")
 
-// Close clears any pending channels that are not in-flight already, to leave a clean derivation state.
-// Close then marks the remaining current open channel, if any, as "full" so it can be submitted as well.
-// Close does NOT immediately output frames for the current remaining channel:
-// as this might error, due to limitations on a single channel.
-// Instead, this is part of the pending-channel submission work: after closing,
-// the caller SHOULD drain pending channels by generating TxData repeatedly until there is none left (io.EOF).
-// A ErrPendingAfterClose error will be returned if there are any remaining pending channels to submit.
-func (s *channelManager) Close() error {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if s.closed {
-		return nil
+// Requeue rewinds blocks back from the current channel
+// and sets the defaultCfg.
+// Should only be called when the channel in question
+// exists and has not started to submit data.
+func (s *channelManager) Requeue(newCfg ChannelConfig) {
+	// Remove the current channel from the back of the queue
+	channelToDiscard := s.channelQueue[len(s.channelQueue)-1]
+	if channelToDiscard != s.currentChannel {
+		panic("current channel is not at the back of the channel queue")
 	}
+	s.channelQueue = s.channelQueue[:len(s.channelQueue)-1]
 
-	s.closed = true
-	s.log.Info("Channel manager is closing")
-
-	// Any pending state can be proactively cleared if there are no submitted transactions
-	for _, ch := range s.channelQueue {
-		if ch.NoneSubmitted() {
-			s.log.Info("Channel has no past or pending submission - dropping", "id", ch.ID())
-			s.removePendingChannel(ch)
-		} else {
-			s.log.Info("Channel is in-flight and will need to be submitted after close", "id", ch.ID(), "confirmed", len(ch.confirmedTransactions), "pending", len(ch.pendingTransactions))
-		}
+	if len(channelToDiscard.channelBuilder.blocks) > 0 {
+		// This is usually true, but there is an edge case
+		// where a channel timed out before any blocks got added.
```

Comment:
Should we add a log for the edge case ( `else { log }`)?

Reply:
Yes we could put a debug level log in for "discarded channel had no blocks").